VHbbAnalysis/Heppy/python/AdditionalBoost.py

```diff
@@ -1,87 +1,485 @@
+                       muonTagInfos, 
+                       elecTagInfos, 
+                       ipTagInfo, 
+                       svTagInfo,
+                       njettiness,
+                       maxSVDeltaRToJet):
+    DEBUG = False
+
+    z_ratio          = -1.
+    tau_dot          = -1.
+    tau_21           = -1.
+    SV_pt_0          = -1.
+    SV_mass_0        = -1.
+    SV_EnergyRatio_0 = -1.
+    SV_EnergyRatio_1 = -1.
+    tau_21           = -1.
+    contSV           = 0
+    vertexNTracks    = 0
+
+    # Re-calculate N-subjettiness independently and get axees
+    fjParticles = ROOT.std.vector("fastjet::PseudoJet")()
+    for dau in jet.daughterPtrVector():
+        if dau.isNonnull() and dau.isAvailable():
+            fjParticles.push_back( ROOT.fastjet.PseudoJet(dau.px(), dau.py(), dau.pz(), dau.energy()))
+
+    tau1_new = njettiness.getTau(1, fjParticles)
+    tau2_new = njettiness.getTau(2, fjParticles)
+    currentAxes = njettiness.currentAxes()
+
+    if not tau1_new == 0:
+        tau_21 = tau2_new/tau1_new
+
+    selectedTracks = ipTagInfo.selectedTracks()
+    trackSize = len(selectedTracks)
+    vertexRef = ipTagInfo.primaryVertex()
+
+    allKinematics = ROOT.reco.TrackKinematics()
+
+    if DEBUG:
+        print "trackSize=", trackSize
+
+    for itt in range(trackSize):
+
+        ptrack = ROOT.reco.btag.toTrack(selectedTracks[itt])
+        ptrackRef = selectedTracks[itt]
+
+        track_PVweight = 0.
+        pcand = ptrackRef.get()
+        if (pcand.fromPV() == ROOT.pat.PackedCandidate.PVUsedInFit):
+            track_PVweight = 1.
+
+        if (track_PVweight>0.):
+            if DEBUG:
+                print "Adding track", itt, "with weight", track_PVweight
+            allKinematics.add(ptrack, track_PVweight)
+    # End of loop over tracks
+
+    jetDir = jet.momentum().Unit()
+
+    if DEBUG:
+        print  "JetDir = ", jetDir.X(), jetDir.Y(), jetDir.Z()
+
+    #std::map<double, size_t> VTXmass;
+    # Just use a Python dict insteas of map
+
+    VTXmass = {}
+
+    for vtx in range(svTagInfo.nVertices()):
+
+        vertexNTracks += (svTagInfo.secondaryVertex(vtx)).numberOfSourceCandidatePtrs()
+        flightDir = svTagInfo.flightDirection(vtx);
+              
+        if DEBUG:
+            print "vtx = ", vtx, "flightDir = ", flightDir.x(), flightDir.y(), flightDir.z()
+
+        if ( deltaR2(flightDir, jetDir)<(maxSVDeltaRToJet*maxSVDeltaRToJet)):
+            if DEBUG:
+                print "pass"
+            contSV += 1
+            VTXmass[svTagInfo.secondaryVertex(vtx).p4().mass()]=vtx
+
+    # End of loop over vertices
+
+    cont=0
+
+    # GlobalVector is defined DataFormats/GeometryVector/interface/GlobalVector.h
+    # typedef Global3DVector GlobalVector;
+    # typedef Vector3DBase< float, GlobalTag>    Global3DVector;
+    # Since there is no dict for GlobalVector build it directly from the template
+    flightDir_0 = ROOT.Vector3DBase(float, ROOT.GlobalTag)()
```

Comment:
@gpetruc @Dr15Jones do you have any clue why this line (and the next 4 lines), introduce a leak of around 45Kb/event?